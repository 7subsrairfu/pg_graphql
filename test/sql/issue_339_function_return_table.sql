begin;
    create table public.account(
        id int primary key
    );

    create function public._computed(rec public.account)
        returns table ( id int )
        immutable
        strict
        language sql
    as $$
        select 2 as id;
    $$;

    insert into account(id) values (1);

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
                name
            }
          }
        }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection {
                edges {
                  node {
                    id
                    computed
                  }
                }
              }
            }
        $$)
    );

rollback;
