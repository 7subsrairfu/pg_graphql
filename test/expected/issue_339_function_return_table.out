begin;
    create table public.account(
        id int primary key
    );
    create function public._computed(rec public.account)
        returns table ( id int )
        immutable
        strict
        language sql
    as $$
        select 2 as id;
    $$;
    insert into account(id) values (1);
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
                name
            }
          }
        }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "__type": {                 +
             "kind": "OBJECT",       +
             "fields": [             +
                 {                   +
                     "name": "nodeId"+
                 },                  +
                 {                   +
                     "name": "id"    +
                 }                   +
             ]                       +
         }                           +
     }                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection {
                edges {
                  node {
                    id
                    computed
                  }
                }
              }
            }
        $$)
    );
                            jsonb_pretty                             
---------------------------------------------------------------------
 {                                                                  +
     "data": null,                                                  +
     "errors": [                                                    +
         {                                                          +
             "message": "Unknown field 'computed' on type 'Account'"+
         }                                                          +
     ]                                                              +
 }
(1 row)

rollback;
